driver-core/src/main/com/mongodb/client/model/Filters.java

```diff
@@ -1023,31 +1023,12 @@ private void addClause(final BsonDocument document, final Map.Entry<String, Bson
             } else if (document.size() == 1 && document.keySet().iterator().next().equals("$and")) {
                 document.get("$and").asArray().add(new BsonDocument(clause.getKey(), clause.getValue()));
             } else if (document.containsKey(clause.getKey())) {
-                if (document.get(clause.getKey()).isDocument() && clause.getValue().isDocument()) {
-                    BsonDocument existingClauseValue = document.get(clause.getKey()).asDocument();
-                    BsonDocument clauseValue = clause.getValue().asDocument();
-                    if (keysIntersect(clauseValue, existingClauseValue)) {
-                        promoteRenderableToDollarForm(document, clause);
-                    } else {
-                        existingClauseValue.putAll(clauseValue);
-                    }
-                } else {
-                    promoteRenderableToDollarForm(document, clause);
-                }
+                promoteRenderableToDollarForm(document, clause);
```

Comment:
I think this can get even simpler.  If we always want to generate $and, and don't care about pulling up nested $and, then this whole method can be deleted and the `toBsonDocument` method can just be:

            BsonArray clauses = new BsonArray();
            for (Bson filter : filters) {
                clauses.add(filter.toBsonDocument(documentClass, codecRegistry));
            }

            return new BsonDocument("$and", clauses);

This will requires a few tests to change, but it should work properly.

Reply:
That is very simple! I have updated `toBsonDocument` to use your code.

Reply:
Only a few tests had to be updated to use `$and`.